index-providers/modeshape-elasticsearch-index-provider/src/main/java/org/modeshape/jcr/index/elasticsearch/EsIndexColumn.java

```diff
@@ -0,0 +1,254 @@
+                break;
+            case LONG:
+            case DECIMAL:
+                builder.startObject(name).field("type", "long").endObject();
+                break;
+            case DOUBLE:
+                builder.startObject(name).field("type", "double").endObject();
+                break;
+            default:
+                builder.startObject(name).field("type", "string")
+                        .field("analyzer", "whitespace").endObject();
+                break;
+        }
+    }
+
+    /**
+     * Converts representation of the given value using type conversation rules
+     * between JCR type of this column and Elasticsearch core type of this column.
+     * 
+     * @param value value to be converted.
+     * @return converted value.
+     */
+    protected Object columnValue(Object value) {
+        switch (type) {
+            case PATH:
+            case NAME:
+            case STRING:
+            case REFERENCE:
+            case SIMPLEREFERENCE:
+            case WEAKREFERENCE:
+            case URI:
+                return valueFactories.getStringFactory().create(value);
+            case DATE :
+                return ((DateTime) value).getMilliseconds();
+        }
+        return value;
+    }
+
+    protected String stringValue(Object value) {
+        return valueFactories.getStringFactory().create(value);
+    }
+    
+    /**
+     * Converts array of values.
+     * @param value
+     * @return 
+     */
+    protected Object[] columnValues(Object[] value) {
+        Object[] res = new Object[value.length];
+        for (int i = 0; i < value.length; i++) {
+            res[i] = columnValue(value[i]);
+        }
+        return res;
+    }
+    
+    /**
+     * Converts given value to the value of JCR type of this column.
+     * 
+     * @param value value to be converted.
+     * @return converted value.
+     */
+    protected Object cast(Object value) {
+        switch (type) {
+            case STRING :
+               return valueFactories.getStringFactory().create(value);
+            case LONG :
+                return valueFactories.getLongFactory().create(value);
+            case NAME :
+                return valueFactories.getNameFactory().create(value);
+            case PATH :
+                return valueFactories.getPathFactory().create(value);
+            case DATE :
+                return valueFactories.getDateFactory().create(value);
+            case BOOLEAN :
+                return valueFactories.getBooleanFactory().create(value);
+            case URI :
+                return valueFactories.getUriFactory().create(value);
+            case REFERENCE :
+                return valueFactories.getReferenceFactory().create(value);
+            case SIMPLEREFERENCE :
+                return valueFactories.getSimpleReferenceFactory().create(value);
+            case WEAKREFERENCE :
+                return valueFactories.getWeakReferenceFactory().create(value);
+        }
```

Comment:
Not having the switch include all possible values is very much preferred, and using the `default` case is a secondary option. However, doing neither is IMO bad form since it's implicitly providing a default case and not using Java's built-in `default` case. Yes, you'll need to throw an exception as the last line of the method to signal that getting there should never happen; even the ugliness of throwing an exception is better than an incomplete switch.